### DISCUSSION TITLE
Taliwindcss with next.js SSR

### DISCUSSION THREAD

**Discussion by MustafaWael** [tailwindlabs/tailwindcss#5083] [Help] [answered]
Hi, i'm using **tailwindcss** with **next.js**, and installed it with `"npx create-next-app -e with-tailwindcss my-project"`

But styles not rendered on the server after next.js build
Styles not shown in network chrome's dev tools

So what should i do?

**Comment by simonswiss** [comment, marked as answer]
Hey!

I copied your command above, pasted it in my terminal, and everything worked out of the box:

![image](https://user-images.githubusercontent.com/485747/127611261-836fe0cc-a460-44b3-ac00-e2416ea2b5de.png)

That's how it looks after running `npm run build` and `npm start`.

If you mean that you cannot see the styles in the `preview` tab of the `network` tab of devtools... That's something that I see a lot, and that really means that styles are not loaded on the server.

![image](https://user-images.githubusercontent.com/485747/127612415-3000b9c9-2784-421d-b2f1-fdcf5f67d690.png)

You can see the CSS styles are also loaded (via `<link>` tag in the `<head>`:

![image](https://user-images.githubusercontent.com/485747/127612668-d0ef0ae1-0736-45c0-b50c-c792498e16e2.png)

I don't know *why* the styles are not showing in the network tab, but the same happens with many NextJS websites.

Vercel website? yep. Tailwind CSS docs? yep.

In all cases, there is absolutely no flash of unstyled content, and the styles are actually rendered on the server - they just don't show up on the preview which doesn't wait for the CSS to come back (the browser does though).

From my understanding, the styles will show up in the preview _only_ if they are inlined in the `head` (and do not require a network request). If you want to handle that, you are probably looking at using a Custom Document:

https://nextjs.org/docs/advanced-features/custom-document

Hope it helps!
